src/app/components/kudos-challenge-new/kudos-challenge-new.component.js

```diff
@@ -0,0 +1,132 @@
+                    toastr.success('You accepted ' + val.data.creator + ' challenge');
+                    removeElement(index);
+                    Resources.setUserAvailableKudos(Resources.getUserAvailableKudos() - val.data.amount);
+                    Resources.getOngoingChallenges().push(val.data);
+                    getChallengeParticipatedList();
+                    refreshList();
+                })
+            } else toastr.error('You only have ' + ' ' + acornPlural(userAvailableKudos) +
+                '. To accept challenge, you must have at least ' + kudos);
+        }
+
+        function cancelChallenge(index) {
+            var challengeId = $httpParamSerializer({
+                id: $scope.challengeList[index].id
+            });
+            console.log(challengeId);
+            console.log(index);
+            Challenges.cancel(challengeId).then(function (val) {
+                Resources.setUserAvailableKudos(Resources.getUserAvailableKudos() + val.data.amount);
+                Resources.getNewChallenges().splice(index, 1);
+                Resources.getCompletedChallenges().push(val.data);
+                console.log(Resources.getNewChallenges());
+                $scope.challengeList = Resources.getNewChallenges();
+
+                toastr.success("Challenge canceled");
+            });
+        }
+
+        function declineChallenge(id, index) {
+            requestData = $httpParamSerializer({
+                id: id
+            });
+            KudosChallengeNewService.decline(requestData).then(function (val) {
+                toastr.info('You declined ' + val.data.creator + ' challenge');
+                removeElement(index);
+                getChallengeParticipatedList();
+                refreshList();
+            })
+        }
+
+        function removeElement(index){
+            if ($scope.challengeList[0]){
+                $scope.challengeList.splice(index, 1);
+            }
+        }
+
+        function refreshList(){
+            $scope.id = false;
+            $scope.challengeList = Resources.getNewChallenges();
+            if ($scope.challengeList[0]) {
+                $scope.id = true;
+            }
+        }
+        
+        function convertDate(val){
+            if (val) {
+                val = val.split(":");
+                val = val[0] + ":" + val[1];
+                return val;
+            }
+        }
+
+        function doesDateExist(index) {
+            return Resources.getNewChallenges()[index].finishDate == null;
+        }
+
+    };
+
+    KudosChallengeNewController.$inject = ['$httpParamSerializer', '$scope', 'KudosChallengeNewService', 'Challenges', 'Resources'];
+
+    angular.module('myApp.components.challengeNew', [])
+
+        .directive('kudosChallengeNew', function () {
+            return {
+                controller: 'KudosChallengeNewController',
+                restrict: 'E',
+                scope: false,
```

Comment:
If you are creating a directive, though I'm certain this could be converted to `angular.component`, it's probably best to use an isolated scope, since you don't want scope pollution by accident and you already have a standalone controller for this. I've yet to find an example of a directive where inheriting the scope from a parent is the best solution of scope initiation.